xarray/core/extension_array.py

```diff
@@ -101,9 +100,9 @@ def __post_init__(self):
         # This does not use the UNSUPPORTED_EXTENSION_ARRAY_TYPES whitelist because
         # we do support extension arrays from datetime, for example, that need
         # duck array support internally via this class.
-        if isinstance(self.array, pd.arrays.NumpyExtensionArray):
+        if not is_allowed_extension_array(self.array):
             raise TypeError(
-                "`NumpyExtensionArray` should be converted to a numpy array in `xarray` internally."
+                "`NumpyExtensionArray` or string dtype should be converted to a numpy array in `xarray` internally."
```

Comment:
```suggestion
                f"{self.array.dtype!r} should be converted to a numpy array in `xarray` internally."
```

Does this work or do we have to protect against `dtype` not existing?